src/signature.rs

```diff
@@ -158,6 +158,13 @@ impl SignedRequest {
 	    // execute the damn request already
 	    let client = Client::new();
 	    let result = client.request(hyper_method, &final_uri).headers(hyper_headers).body(&payload).send().unwrap();
+
```

Comment:
Some kind of proper logging framework for Rust must exist by now.  We should look into that.